scripts/data_prep/convert_delta_to_json.py

```diff
@@ -0,0 +1,490 @@
+
+    try:
+        ans = run_query(f'SELECT COUNT(*) FROM {tablename}', method, cursor,
+                        sparkSession)
+        nrows = [row.asDict() for row in ans][0].popitem()[1]  # pyright: ignore
+        log.debug(f'total_rows = {nrows}')
+    except Exception as e:
+        raise RuntimeError(
+            f'Error in get total rows from {tablename}. Restart sparkSession and try again'
+        ) from e
+
+    try:
+        ans = run_query(f'SHOW COLUMNS IN {tablename}', method, cursor,
+                        sparkSession)
+        columns = [row.asDict().popitem()[1] for row in ans]  # pyright: ignore
+        order_by = columns[0]
+        columns_str = ','.join(columns)
+        log.info(f'order by column {order_by}')
+    except Exception as e:
+        raise RuntimeError(
+            f'Error in get columns from {tablename}. Restart sparkSession and try again'
+        ) from e
+
+    if method == 'dbconnect' and sparkSession is not None:
+        log.info('processes = ', processes)
+        df = sparkSession.table(
+            tablename)  # "main.tpcds_sf100_delta.store_sales")
+
+        # Running the query and collecting the data as arrow or json.
+        signed, _, _ = df.collect_cf('arrow')  # pyright: ignore
+        log.info(f'len(signed) = {len(signed)}')
+
+        args = get_args(signed, json_output_path, columns)
+
+        # Stopping the SparkSession to avoid spilling connection state into the subprocesses.
+        sparkSession.stop()
+
+        with ProcessPoolExecutor(max_workers=processes) as executor:
+            list(executor.map(download_starargs, args))
+
+    elif method == 'dbsql' and cursor is not None:
+        for start in range(0, nrows, batch_size):
+            log.info('start = ', start)
+            end = min(start + batch_size, nrows)
+            fetch_data(method, cursor, sparkSession, start, end, order_by,
+                       tablename, columns_str, json_output_path)
+
+    if cursor is not None:
+        cursor.close()
+
+
+def fetch_DT(args: Namespace) -> None:
+    """Fetch UC Delta Table to local as jsonl."""
+    log.info(f'Start .... Convert delta to json')
+
+    obj = urllib.parse.urlparse(args.json_output_path)
+    if obj.scheme != '':
+        raise ValueError(
+            f'Check the json_output_path and verify it is a local path!')
+
+    if os.path.exists(args.json_output_path):
+        if not os.path.isdir(args.json_output_path) or os.listdir(
+                args.json_output_path):
+            raise RuntimeError(
+                f'A file or a folder {args.json_output_path} already exists and is not empty. Remove it and retry!'
+            )
+
+    os.makedirs(args.json_output_path, exist_ok=True)
+
+    log.info(f'Directory {args.json_output_path} created.')
+
+    method = 'dbsql'
+    dbsql = None
+    sparkSession = None
+
+    w = WorkspaceClient()
+    res = w.clusters.get(cluster_id=args.cluster_id)
```

Comment:
FYI this won't work if the `cluster_id` is a serverless `cluster_id`, see [my notebook](https://e2-dogfood.staging.cloud.databricks.com/?o=6051921418418893#notebook/187185339621026/command/187185339621075). We can pass in `None` or a special `serverless` string to recognize a serverless cluster and skip this check. Let me know how you want to default this and I can update Mcloud / the SDK accordingly @XiaohanZhangCMU 

cc @panchalhp-db

Reply:
@nancyhung good catch. Can you check the new logic? Let's use "serverless" as the magic word to skip the check.